specs/WCT/Primitives/WrapPanel.md

```diff
@@ -0,0 +1,213 @@
+then wraps the Green box around to the next line due to the Width constraint on the WrapPanel:
+
+```xml
+    <controls:WrapPanel Width="132" HorizontalSpacing="16">
+        <Rectangle Fill="Red" Width="44" Height="44"/>
+        <Rectangle Fill="Blue" Width="44" Height="44"/>
+        <Rectangle Fill="Green" Width="44" Height="44"/>
+        <Rectangle Fill="Orange" Width="44" Height="44"/>
+    </controls:WrapPanel>
+```
+
+The result looks like this:
+
+![WrapPanel HorizontalSpacing example](images/WrapPanelHorizontalSpacing.png)
+
+## WrapPanel.Orientation property
+
+Gets or sets the orientation of the WrapPanel. **Horizontal** means that child controls will be added 
+horizontally until the width of the panel is reached, then a new row is added to add new child controls.
+**Vertical** means that children will be added vertically until the height of the panel is reached, then 
+a new column is added. Default is **Horizontal**.
+
+## WrapPanel.Padding property
+
+Gets or sets the distance between the border and its child object.
+
+## WrapPanel.StretchChild property
+
+Gets or sets how the last child item in the WrapPanel is laid out. When set to **Last**, the final child 
+can stretch and be given all the rest of the available space for the row (when Orientation is 
+**Horizontal**) or column (when Orientation is **Vertical**). Defaults to **None** where no changes to 
+default behavior layout are provided.
+
+In the following example, the last Orange rectangle will be constrained and pushed to the 2nd row. 
+Normally if given the same Width/Height as the others, it'd sit just below the Red rectangle to the left. 
+Instead, we can remove it's Width and have it fill the rest of the row by setting `StretchChild` to **Last**:
+
+```xml
+    <controls:WrapPanel Width="132" StretchChild="Last">
+        <Rectangle Fill="Red" Width="44" Height="44"/>
+        <Rectangle Fill="Blue" Width="44" Height="44"/>
+        <Rectangle Fill="Green" Width="44" Height="44"/>
+        <Rectangle Fill="Orange" Height="44" />
+    </controls:WrapPanel>
+```
+
+The result looks like this:
+
+![WrapPanel StretchChild example](images/WrapPanelStretchChild.png)
+
+
+## Other WrapPanel members
+
+| Name | Description |
+|-|-|
+| VerticalSpacing | Gets or sets a uniform Vertical distance (in pixels) between items when Orientation is set to Vertical, or between rows of items when Orientation is set to Horizontal. (defaults to 0) |
+
+# API Details
+
+```c# (but really MIDL3)
+[contract(Microsoft.UI.Xaml.WinUIContract, )]
+[webhosthidden]
+namespace Microsoft.UI.Xaml.Controls
+{
+  unsealed runtimeclass WrapPanel
+      : Microsoft.UI.Xaml.Controls.Panel
+  {
+      [method_name("CreateInstance")] WrapPanel();
+
+      Double HorizontalSpacing;
+      Orientation Orientation;
+      Thickness Padding;
+      StretchChild StretchChild;
+      Double VerticalSpacing;
+
+      static Microsoft.UI.Xaml.DependencyProperty HorizontalSpacingProperty { get; };
```

Comment:
Let's say I want to have a 6 items in my panel. They're all buttons whose minimum size is 30.  My wrap panel is autoresized to be width 360 (2x the required size to display all six buttons in a row.)  With `StretchChild=Last`, I see (for widths, in a single row) `30, 30, 30, 30, 30, 210`.

Is there a way to have all the items in the row expand to fit equal amounts, so I'd see `60, 60, 60, 60, 60, 60`? Or if my width was 240, have it produce `{60, 60, 60, 60},{120, 120}`?